src/phoenix/db/bulk_inserter.py

```diff
@@ -104,13 +111,20 @@ async def _bulk_insert(self) -> None:
             # Spans should be inserted before the evaluations, since an evaluation
             # insertion will fail if the span it references doesn't exist.
             if spans_buffer:
-                await self._insert_spans(spans_buffer)
+                project_rowids = await self._insert_spans(spans_buffer)
+                updated_project_rowids.update(project_rowids)
                 spans_buffer = None
             if evaluations_buffer:
-                await self._insert_evaluations(evaluations_buffer)
+                project_rowids = await self._insert_evaluations(evaluations_buffer)
+                updated_project_rowids.update(project_rowids)
                 evaluations_buffer = None
+            if updated_project_rowids:
+                for project_rowid in updated_project_rowids:
+                    self._last_inserted_at[project_rowid] = datetime.now(timezone.utc)
+                updated_project_rowids.clear()
 
-    async def _insert_spans(self, spans: List[Tuple[Span, str]]) -> None:
+    async def _insert_spans(self, spans: List[Tuple[Span, str]]) -> Set[ProjectRowId]:
```

Comment:
the return signature could be maybe made a better shape like an object that encapsulates the updates. That this updates contains the "touched" projects.

Otherwise the return signature is a tad non-intuitive. I'd imagine a list of spans or a list of span ids if I wasn't reading carefully